src/mds/Beacon.cc

```diff
@@ -510,9 +510,19 @@ void Beacon::notify_health(MDSRank const *mds)
         MDSHealthMetric m(MDS_HEALTH_CLIENTS_LAGGY, HEALTH_WARN, css->strv());
         laggy_clients_metrics.emplace_back(std::move(m));
       }
-      auto&& m = laggy_clients_metrics;
-      health.metrics.insert(std::end(health.metrics), std::cbegin(m),
-                            std::cend(m));
+
+      if (laggy_clients.size() <= (size_t)g_conf()->mds_health_summarize_threshold) {
```

Comment:
In your first commit you just reverted some old code, and then no where will insert items to the `laggy_clients`, right ?

Reply:
still in evict_cap_revoke_non_responders().

Reply:
The same, I didn't find it's used any where:

```
xiubli:src$ grep defer_client_eviction_on_laggy_osds ./* -rn
./common/options/mds.yaml.in:1560:- name: defer_client_eviction_on_laggy_osds
xiubli:src$ 

```

Reply:
```
void Server::evict_cap_revoke_non_responders() {
  if (!cap_revoke_eviction_timeout) {
    return;
  }

  auto&& to_evict = mds->locker->get_late_revoking_clients(cap_revoke_eviction_timeout);
  // don't evict client(s) if osds are laggy
  bool defer_client_eviction =
  g_conf().get_val<bool>("defer_client_eviction_on_laggy_osds")
  && mds->objecter->with_osdmap([](const OSDMap &map) {
    return map.any_osd_laggy(); })
  && to_evict.size();
```

Reply:
So strange, I just cherry-picked your these 2 commit basing on the upstream ???